### DISCUSSION TITLE
Deprecated Method setUndefinedTargetValidatesAllSubjects in RDF4J 4.3.6

### DISCUSSION THREAD

**Discussion by vallerap** [eclipse-rdf4j/rdf4j#4808] [Q&A] [answered]
Hello,

I'm currently in the process of upgrading my project from RDF4J version 3.7.7 to version 4.3.6, and I've encountered an issue related to the deprecated method setUndefinedTargetValidatesAllSubjects. In the previous version (3.7.7), this method was crucial for my project, but it has been deprecated in the latest version.

I'm facing difficulties in achieving the same results as before due to this deprecation. 
Is anyone aware of alternative approaches or workarounds to achieve the same result in RDF4J 4.3.6?

**Comment by hmottestad** [comment]
We now support DASH which should solve this: https://datashapes.org/dash#AllObjectsTarget

Just remember to enable the config for DASH in the ShaclSail.

**Reply by hmottestad** [reply]
We don’t support all of DASH mind you.

**Reply by vallerap** [reply]
Hi @hmottestad  :) thanks for the fast reply

Is it sufficient to set `shaclSail.setDashDataShapes(true);`?
then I guess the validation occurs on the graphs defined by `shaclSail.setShapesGraphs(graphs)` ?

**Reply by hmottestad** [reply, marked as answer]
Yes, that should be sufficient. 

As for shapes graph, see our documentation on how that works here: https://rdf4j.org/documentation/programming/shacl/#shapes-graph

**Reply by vallerap** [reply]
Hi and thank you again,

... I'm not sure if I understand correctly but it seems that I can achieve the same result with:
`shaclSail.setShapesGraphs(Set.of(RDF4J.SHACL_SHAPE_GRAPH, DASH.AllObjectsTarget));` ?

**Reply by hmottestad** [reply]
Here you can see an example: https://github.com/eclipse-rdf4j/rdf4j/blob/86cbf5b9c7eb2712cb01af8821cdbb92a6a2a7ae/core/sail/shacl/src/test/resources/test-cases/datatype/allObjects/shacl.trig
